noir-projects/aztec-nr/aztec/src/oracle/notes.nr

```diff
@@ -175,33 +146,30 @@ where
         limit,
         offset,
         status,
+        MAX_NOTES,
+        NOTE_PCKD_LEN + RETRIEVED_NOTE_OVERHEAD,
     );
-    let num_notes = fields[0] as u32;
-    let contract_address = AztecAddress::from_field(fields[1]);
-
-    let mut opt_notes = [Option::none(); MAX_NOTES];
-    for i in 0..opt_notes.len() {
-        if i < num_notes {
-            // lengths named as per typescript.
-            let return_header_length: u32 = 2; // num_notes & contract_address.
-            let extra_preimage_length: u32 = 2; // nonce & note_hash_counter.
-            let read_offset: u32 = return_header_length + i * (N + extra_preimage_length);
-
-            let maybe_nonce = fields[read_offset];
-            let maybe_note_hash_counter = fields[read_offset + 1] as u32;
-            let packed_note = array::subarray(fields, read_offset + 2);
-
-            let note = Note::unpack(packed_note);
-            let retrieved_note = RetrievedNote {
-                note,
-                contract_address,
-                metadata: NoteMetadata::from_raw_data(maybe_note_hash_counter != 0, maybe_nonce),
-            };
-
-            opt_notes[i] = Option::some(retrieved_note);
+
+    let mut notes = BoundedVec::new();
+    for i in 0..packed_retrieved_notes.len() {
+        let packed_retrieved_note = packed_retrieved_notes.get(i);
```

Comment:
In a PR up the stack I will simplify this by moving the contents of this loop to an impl of a Packable trait for RetrievedNote